far/string_utils.hpp

```diff
@@ -180,28 +180,54 @@ class icase_searcher final: public i_searcher
 	exact_searcher m_Searcher;
 };
 
+namespace detail
+{
+	class fuzzy_searcher_impl
+	{
+	public:
+		string_view normalize(string_view Str);
+		std::optional<std::pair<size_t, size_t>> find_in(const i_searcher& searcher, string_view Haystack, bool Reverse);
+
+	private:
+		string m_Result, m_Intermediate;
+		std::vector<WORD> m_Types;
+	};
+}
+
+template <typename Searcher>
+#ifdef __cpp_concepts
+	requires std::is_constructible_v<Searcher, string_view, bool>
```

Comment:
Is it really needed? It doesn't look like we need to constraint the template here to solve some overload resolution issues or to help the users avoid confusion when instantiating it with arbitrary types on a daily basis.

If it is needed anyway, please see the `REQUIRES()` macro.

Reply:
I removed redundant `requires`.